### PR TITLE
Fixed Term Count Issue after Event Deletion

### PR DESCRIPTION
**Pull request by hoseinrafiei**
Fixes https://github.com/eventespresso/event-espresso-core/issues/4027 issue.

### CODE REVIEW

File: core/db_models/EEM_Term_Relationship.model.php [hunk 1]
```diff
@@ -218,6 +224,18 @@ public function delete($query_params, $allow_blocking = true)
...
+     */
+    public function delete_permanently($query_params, $allow_blocking = true)
+    {
+        $count = parent::delete_permanently($query_params, $allow_blocking);
+        if ($count) {
+            $this->update_term_taxonomy_counts();
+        }
+        return $count;
+    }
+
```
Comment (hoseinrafiei) [inline core/db_models/EEM_Term_Relationship.model.php:238]: Exact fix.

File: core/db_models/EEM_Term_Relationship.model.php [hunk 2]
```diff
@@ -13,22 +13,22 @@ class EEM_Term_Relationship extends EEM_Base
...
+     * @throws EE_Error
      */
     protected function __construct($timezone = null)
     {
-        $this->singular_item = esc_html__('Term Relationship', 'event_espresso');
-        $this->plural_item = esc_html__('Term Relationships', 'event_espresso');
-        $this->_tables = array(
+        $this->singular_item       = esc_html__('Term Relationship', 'event_espresso');
+        $this->plural_item         = esc_html__('Term Relationships', 'event_espresso');
+        $this->_tables             = [
```
Comment (tn3rb) [inline core/db_models/EEM_Term_Relationship.model.php:26]: thnx for formatting 🙇🏻‍♂️

File: core/db_models/EEM_Term_Relationship.model.php [hunk 3]
```diff
@@ -147,23 +150,23 @@ public function update_term_taxonomy_counts($term_taxonomy_id = null)
...
-        $rows_affected = $this->_do_wpdb_query(
+        return $this->_do_wpdb_query(
             'query',
-            array(
+            [
                 $query,
-            )
+            ]
         );
-        return $rows_affected;
```
Comment (tn3rb) [inline core/db_models/EEM_Term_Relationship.model.php:156]: 👍🏻

**Review by tn3rb — APPROVED** [review]
# Looks Great 🏆 🏇🏻

**Comment by Saam01** [comment]
Seems to be working fine now.

**Pull request closed without merging**